### ISSUE TITLE
Oga Adapter

### ISSUE THREAD

**Issue by bryanp** [premailer/premailer#283]
Hello! We're trying to remove Nokogiri as a dependency in [Pakyow](https://github.com/pakyow/pakyow) (using [Oga](https://github.com/YorickPeterse/oga) instead). The problem is that our mailer lib uses Premailer, which is currently setup to use the Nokogiri adapter. To use Oga throughout the project we'd like contribute an appropriate Premailer adapter.

Has this been attempted? If not, any tips before getting started?

**Comment by akzhan** [comment]
You can replace Nokogiri with Nokogumbo one.

**Comment by akzhan** [comment]
Or you can to provide Oga adapter to me merged into.

**Comment by bryanp** [comment]
@akzhan Unfortunately the adapter interface is incredibly tied to Nokogiri (or libraries like Nokogumbo that follow its interface). We made it work in pakyow, but it's not something I'd want to see in Premailer as-is. Doing it correctly would be a larger effort of redesigning the adapter interface. Willing to help, but is redesigning things at this level something that you all are open to doing?

**Comment by akzhan** [comment]
Yes, any good contribution that did project better will be landed

Вт, 28 июня 2016 г. в 0:59, Bryan Powell <notifications@github.com>:
